infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/infrastructure/plugins/CorfuReplicationClusterManagerBaseAdapter.java

```diff
@@ -0,0 +1,52 @@
+package org.corfudb.infrastructure.logreplication.infrastructure.plugins;
+
+import lombok.Getter;
+
+import lombok.extern.slf4j.Slf4j;
+import org.corfudb.infrastructure.logreplication.infrastructure.CorfuReplicationDiscoveryServiceAdapter;
+import org.corfudb.infrastructure.logreplication.proto.LogReplicationClusterInfo.TopologyConfigurationMsg;
+
+/***
+ * This is the base class for CorfuReplicationSiteManager and implements the basic functionality.
+ * Any SiteMangerImplementation should extend this class or implements the interface.
+ *
+ */
+@Slf4j
+public abstract class CorfuReplicationClusterManagerBaseAdapter implements CorfuReplicationClusterManagerAdapter {
```

Comment:
Is this abstract class used to guarantee that all cluster managers have corfuReplicationDiscoveryService and topologyConfig?

Reply:
yes.